c2dataviewer/control/scopecontroller.py

```diff
@@ -154,6 +158,52 @@ def __flatten_dict(dobj, kprefixs=[]):
             else:
                 yield sep.join(kprefixs + [k]), v
 
+    def execute_later(instruction : Callable[[], None]) -> None :
+        '''
+        Executes instruction after event pile is emptied, avoiding SegFaults. See QtCore's documentation.
+
+        :param instruction: The function to be executed.
+        '''
+        QtCore.QTimer.singleShot(0, instruction)
+
+    def muted(self, instruction : Callable[[], None]) -> None :
+        '''
+        Disconnects then reconnects signal emited on parameter change, avoiding recursive calls and SegFaults.
+        Should be used each time code modifies a parameter.
+        
+        :param: instruction: The function to be executed.
+        '''
+        # Try to disconnect signal. If already disconnected, TypeError is raised, then execute instruction as it is.
+        try:
+            self.parameters.sigTreeStateChanged.disconnect(self.parameter_change)
+            instruction()
+            self.parameters.sigTreeStateChanged.connect(self.parameter_change)
+        except TypeError:
+            instruction()
+
+    def set_channels_number(self, number : int) -> None :
+        '''
+        Changes the number of channel input sections.
+        
+        :param number: The new number of sections that should be displayed.
+        '''
+        if number > 10 or number < 1 or type(number) is not int:
+            raise ValueError(f'Invalid channel number: {number}. Must be integer between 1 and 10.')
+        previous_number = len(self.channels)
+        if previous_number > number:
+            self.channels = self.channels[: number]
+            while previous_number > number:
+                self.muted(lambda : self.parameters.child('Channel %s' % previous_number).remove())
+                previous_number -= 1
+        else:
+            while previous_number < number:
+                self.muted(lambda : self.parameters.insertChild(self.parameters.child('Statistics'), Configure.new_channel(previous_number + 1, self.color_pattern[previous_number], ['None'], 0)))
```

Comment:
Can you add a comment in the code explaining what this part is doing, I think it is a little tricky since it depends on Statistics being right after the Channel sections.